wgsl/index.bs

```diff
@@ -12382,14 +12385,15 @@ specify the component type; the component type is inferred from the constructor
       <td>
         <xmp highlight=wgsl>@const @must_use fn i32(e : T) -> i32</xmp>
   <tr><td>Parameterization
-      <td>`T` is a [=type/concrete=] [=scalar=] type
+      <td>`T` is a [=scalar=] type
   <tr><td>Description
       <td>Construct an [=i32=] value.
 
       If `T` is [=i32=], this is an identity operation.<br>
       If `T` is [=u32=], this is a reinterpretation of bits (i.e. the result is the unique value in [=i32=] that has the same bit pattern as `e`).<br>
       If `T` is a [[#floating-point-types|floating point type]], `e` is [=scalar floating point to integral conversion|converted=] to [=i32=], rounding towards zero.<br>
```

Comment:
Note to myself: AbstractFloat is now captured by this case directly.  Before, we would use the automatic conversion AbstractFloat --> f32  and then use the rule for f32.

This implies the float number has to keep all 31 mantissa bits when constructing an i32.
for example

```

const big0 = 0x40000000.0;
const big1 = 0x40000001.0;
const big2 = 0x40000002.0;  // these differ only in the 31st bit.
const_assert( i32(big0) < i32(big1));  // true
const_assert( i32(big1) < i32(big2));  // true
```